pkg/deploy/strategy/support/lifecycle.go

```diff
@@ -67,11 +74,24 @@ func NewHookExecutor(client kclient.PodsNamespacer, tags client.ImageStreamTagsN
 // Execute executes hook in the context of deployment. The suffix is used to
 // distinguish the kind of hook (e.g. pre, post).
 func (e *HookExecutor) Execute(hook *deployapi.LifecycleHook, deployment *kapi.ReplicationController, suffix, label string) error {
-	var err error
+	config, err := deployutil.DecodeDeploymentConfig(deployment, e.decoder)
+	if err != nil {
+		return err
```

Comment:
if we fail decode the RC into DC, then I think something terrible is happening for you and we should error out.